Sundials CI failure for python:test_kinetics.KineticsRepeatability.test_pdep_err

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

CI test for `python:test_kinetics.KineticsRepeatability.test_pdep_err` fails for Sundials runners. All other tests are successful. 

Issue was observed while testing #943 - which presumably does not affect Kinetics.
 
**Steps to reproduce**

Presumably any newly created PR will trigger this issue.

**Behavior**

```
======================================================================
FAIL: test_pdep_err (cantera.test.test_kinetics.KineticsRepeatability)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cantera/cantera/build/python/cantera/test/test_kinetics.py", line 419, in test_pdep_err
    ct.Solution('addReactions_err_test.yaml')
cantera._cantera.CanteraError: 
***********************************************************************
CanteraError thrown by addReactions:

***********************************************************************
CanteraError thrown by Plog::validate:

Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'
at P = 32019, T = 500

Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'
at P = 32019, T = 1000

Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'
at P = 1.0132e+05, T = 500

Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'
at P = 1.0132e+05, T = 1000
***********************************************************************

***********************************************************************
CanteraError thrown by Plog::validate:

Invalid rate coefficient for reaction 'CH2CHOO <=> CH3 + CO2'
at P = 1.0132e+07, T = 500
***********************************************************************

***********************************************************************
InputFileError thrown by Kinetics::checkReactionBalance:
Error on line 126 of /home/runner/work/cantera/cantera/build/python/cantera/test/data/addReactions_err_test.yaml:
The following reaction is unbalanced: H2O2 + OH <=> 2 H2O + HO2
  Element    Reactants    Products
  H           3           5
  O           3           4

|  Line |
|   121 | - equation: CH2CHOO <=> CO2 + CH3  # Reaction 2
|   122 |   type: pressure-dependent-Arrhenius
|   123 |   rate-constants:
|   124 |   - {P: 100.0 atm, A: -2510.0, b: 1.41, Ea: 1.442e+04}
|   125 |   - {P: 100.0 atm, A: 4.05e-09, b: 5.14, Ea: 1.048e+04}
>   126 > - equation: H2O2 + OH <=> 2 H2O + HO2  # Reaction 3
            ^
|   127 |   rate-constant: {A: 1.0e+13, b: 0.0, Ea: 1800.0}
***********************************************************************
***********************************************************************


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/work/cantera/cantera/build/python/cantera/test/test_kinetics.py", line 424, in test_pdep_err
    self.assertIn(msg, err_msg_list)
AssertionError: 'at P = 32019, T = 500.0' not found in ['', '***********************************************************************', 'CanteraError thrown by addReactions:', '', '***********************************************************************', 'CanteraError thrown by Plog::validate:', '', "Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'", 'at P = 32019, T = 500', '', "Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'", 'at P = 32019, T = 1000', '', "Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'", 'at P = 1.0132e+05, T = 500', '', "Invalid rate coefficient for reaction 'CH2CHOO <=> CH3O + CO'", 'at P = 1.0132e+05, T = 1000', '***********************************************************************', '', '***********************************************************************', 'CanteraError thrown by Plog::validate:', '', "Invalid rate coefficient for reaction 'CH2CHOO <=> CH3 + CO2'", 'at P = 1.0132e+07, T = 500', '***********************************************************************', '', '***********************************************************************', 'InputFileError thrown by Kinetics::checkReactionBalance:', 'Error on line 126 of /home/runner/work/cantera/cantera/build/python/cantera/test/data/addReactions_err_test.yaml:', 'The following reaction is unbalanced: H2O2 + OH <=> 2 H2O + HO2', '  Element    Reactants    Products', '  H           3           5', '  O           3           4', '', '|  Line |', '|   121 | - equation: CH2CHOO <=> CO2 + CH3  # Reaction 2', '|   122 |   type: pressure-dependent-Arrhenius', '|   123 |   rate-constants:', '|   124 |   - {P: 100.0 atm, A: -2510.0, b: 1.41, Ea: 1.442e+04}', '|   125 |   - {P: 100.0 atm, A: 4.05e-09, b: 5.14, Ea: 1.048e+04}', '>   126 > - equation: H2O2 + OH <=> 2 H2O + HO2  # Reaction 3', '            ^', '|   127 |   rate-constant: {A: 1.0e+13, b: 0.0, Ea: 1800.0}', '***********************************************************************', '***********************************************************************']

----------------------------------------------------------------------

```
<!-- Describe the result of executing the above steps, and how this differs from what you expect to happen. -->

**System information**

- Cantera version: 2.5.0b1 `main`
- OS: Sundials CI runner
- Python/MATLAB/other software versions: N/A

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -16,11 +16,12 @@ on:
 
 jobs:
   ubuntu-multiple-pythons:
-    name: Ubuntu 18.04 with Python ${{ matrix.python-version }}
-    runs-on: ubuntu-latest
+    name: Ubuntu with Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: [ '3.5', '3.6', '3.7', '3.8' ]
+        python-version: [ '3.5', '3.6', '3.7', '3.8', '3.9' ]
+        os: ['ubuntu-18.04', 'ubuntu-20.04']
       fail-fast: false
     steps:
     - uses: actions/checkout@v2
@@ -28,7 +29,7 @@ jobs:
       with:
         submodules: recursive
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
         architecture: x64
@@ -45,10 +46,11 @@ jobs:
 
   macos-multiple-pythons:
     name: macOS with Python ${{ matrix.python-version }}
-    runs-on: macos-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         python-version: [ '3.5', '3.6', '3.7', '3.8' ]
+        os: ['macos-10.15', 'macos-11.0']
       fail-fast: false
     steps:
     - uses: actions/checkout@v2
@@ -56,7 +58,7 @@ jobs:
       with:
         submodules: recursive
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
         architecture: x64
@@ -85,7 +87,7 @@ jobs:
       with:
         submodules: recursive
     - name: Setup python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: '3.8'
         architecture: x64
@@ -116,7 +118,7 @@ jobs:
         with:
           submodules: recursive
       - name: Setup Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
           python-version: '3.8'
           architecture: x64
@@ -174,15 +176,19 @@ jobs:
   run-examples:
     name: Run the Python examples using bash
     runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version: ['3.5', '3.6', '3.7', '3.8', '3.9']
+      fail-fast: false
     steps:
       - uses: actions/checkout@v2
         name: Checkout the repository
         with:
           submodules: recursive
       - name: Setup Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
-          python-version: '3.8'
+          python-version: ${{ matrix.python-version }}
           architecture: x64
       - name: Install Apt dependencies
         run: sudo apt-get install libboost-dev gfortran graphviz liblapack-dev libblas-dev
@@ -221,7 +227,7 @@ jobs:
         name: Checkout the repository
         with:
           submodules: recursive
-      - uses: goanpeca/setup-miniconda@v1
+      - uses: conda-incubator/setup-miniconda@v2
         with:
           auto-update-conda: true
           python-version: 3.8
@@ -231,7 +237,7 @@ jobs:
       - name: Install conda dependencies
         shell: bash -l {0}
         run: |
-          conda install -q sundials=${{ matrix.sundials-ver}} scons numpy ruamel_yaml \
+          conda install -q sundials=${{ matrix.sundials-ver }} scons numpy ruamel_yaml \
           cython libboost fmt eigen yaml-cpp h5py pandas
       - name: Build Cantera
         run: |
@@ -252,7 +258,7 @@ jobs:
       with:
         submodules: recursive
     - name: Setup python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
       with:
         python-version: '3.8'
         architecture: x64
@@ -276,7 +282,7 @@ jobs:
       matrix:
         os: ['windows-2019']
         vs-toolset: ['14.0', '14.2']
-        python-version: [ '3.5', '3.6', '3.7', '3.8' ]
+        python-version: [ '3.5', '3.6', '3.7', '3.8', '3.9' ]
         # Must use windows-2016 image because it installs VS2017 (MSVC 14.1)
         # Scons cannot find MSVC 14.1 when VS2019 is installed
         include:
@@ -292,6 +298,9 @@ jobs:
           - os: 'windows-2016'
             vs-toolset: '14.1'
             python-version: '3.8'
+          - os: 'windows-2016'
+            vs-toolset: '14.1'
+            python-version: '3.9'
       fail-fast: false
     steps:
       - uses: actions/checkout@v2
@@ -299,17 +308,17 @@ jobs:
         with:
           submodules: recursive
       - name: Set Up Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-          architecture: 'x64'
+          architecture: x64
       - name: Install Python dependencies
         run: |
           python -m pip install -U pip 'setuptools>=47.0.0,<48'
           python -m pip install scons pypiwin32 numpy ruamel.yaml cython h5py pandas
       - name: Build Cantera
         run: |
-          scons build -j2 boost_inc_dir=%BOOST_ROOT_1_69_0% debug=n VERBOSE=y python_package=full ^
+          scons build -j2 boost_inc_dir=%BOOST_ROOT_1_72_0% debug=n VERBOSE=y python_package=full ^
           msvc_version=${{ matrix.vs-toolset }} f90_interface=n
         shell: cmd
       - name: Test Cantera

--- a/src/kinetics/RxnRates.cpp
+++ b/src/kinetics/RxnRates.cpp
@@ -103,7 +103,7 @@ void Plog::validate(const std::string& equation)
             if (k < 0) {
                 format_to(err_reactions,
                           "\nInvalid rate coefficient for reaction '{}'\n"
-                          "at P = {:.5g}, T = {}\n",
+                          "at P = {:.5g}, T = {:.1f}\n",
                           equation, std::exp(iter->first), T[i]);
             }
         }

